### PR TITLE
Achievement Get Part 1 - Greentext Achievements

### PR DESCRIPTION
**Pull request by Altoids1**
Hopek and Readystorm are paying me for this.
image

## Overview

Included are 18 of the 88 achievements I'm being paid to add to the game, all of them having to do with greentexting.

There are also unique achievements for the leaders of team antag groups winning, too. Like Eminences get both the default "win as clockie" achievement as well as the "win as Eminence" one.

## Coder Warning

Doing this involved the creation of a new achievement type, ``/datum/achievement/greentext``. All achievements related to greentexting should be put under this type so that the user can also get the generic "get a greentext" achievement, too.

Additionally, ``/datum/antagonist`` got a new var: ``greentext_achieve``, which is the type of an achievement the antag will receive if they greentext as that antag type.

Not all antag types have been converted to the new ``greentext_achieve`` method, due to the excessive copypaste and lack of inheritence when it comes to determining win condition and objective accomplishment.

## Changelog
:cl:  Altoids
rscadd: Gangs, Clownops, Generic antags, EAAs, IAAs, Culties, Clockies, Revs, Sentient Diseases, Revenants, Vampires, Pirates and Blobs now all have unique achievements for greentexting!
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
do my kudzu bounty

**Comment by monster860** [comment]
You should add an achievement for greentexting as a nightmare.

**Comment by Altoids1** [comment]
> You should add an achievement for greentexting as a nightmare.

That'll be $0.34 at the next window please

**Comment by Hopekz** [comment]
> You should add an achievement for greentexting as a nightmare.

LOL is there any way to NOT greentext as nightmare?

**Comment by Altoids1** [comment]
Okay, that should be all the Greentext achievements on my list done, with a couple runtime fixes already committed.

# Ready for Review.

File: code/datums/achievements/achievements.dm [hunk 1]
```diff
@@ -131,3 +137,108 @@
...
+	name = "Grey Goo"
+	desc = "As a Blob complete your objectives"
+	id = GREENTEXT + 10
+
+/datum/achievement/greentext/clownop
+	name = "\"You wouldn't get it\""
+	desc = "As a Clown Operative score a Major or Minor Victory"
+	id = GREENTEXT + 11
+
+/datum/achievement/greentext/internal
```
Comment (alexkar598) [inline code/datums/achievements/achievements.dm:198]: I realise this is a pretty fucking nitpick but can you rename this and the external one to iaa and eaa, internal makes it sound like it's some kind of abstract sub type for internal use

**Review by alexkar598 — CHANGES_REQUESTED** [review]
My small little nitpick and also you have to write a database query to unlock the green text achievement on everybody who already has a green text achievement

File: code/controllers/subsystem/achievements.dm [hunk 2]
```diff
@@ -52,6 +52,8 @@ SUBSYSTEM_DEF(achievements)
 	if(!achievement)
 		log_sql("Achievement [achievementPath] not found in list of achievements when trying to unlock for [C.ckey]")
 		return FALSE
+	if(istype(achievement,/datum/achievement/greentext) && achievementPath != /datum/achievement/greentext)
```
Comment (ghost) [inline code/controllers/subsystem/achievements.dm:55]: bruh

Comment (Fikou) [reply code/controllers/subsystem/achievements.dm:55]: lol

**Pull request merged**